torch/_prims_common/__init__.py

```diff
@@ -612,7 +616,7 @@ def check_same_shape(*args, allow_cpu_scalar_tensors: bool):
         if isinstance(arg, Number):
             continue
         elif isinstance(arg, TensorLike):
-            if allow_cpu_scalar_tensors and is_cpu_scalar_tensor(arg):
+            if allow_cpu_scalar_tensors and is_cpu_or_meta_scalar_tensor(arg):
```

Comment:
This doesn't look right

Reply:
Why is `in_kernel_invocation_manager` needed here https://github.com/pytorch/pytorch/blob/404b8fb1e236979b8997968d9f40b2825245b13a/torch/_subclasses/fake_tensor.py#L871
Because of "in kernel invocation" devices of inputs to this function are all `meta`, not fake cpu/cuda https://github.com/pytorch/pytorch/blob/404b8fb1e236979b8997968d9f40b2825245b13a/torch/_prims/__init__.py#L321

Reply:
Check out https://github.com/pytorch/pytorch/issues/87236

Reply:
cc @eellison, but the idea here is we have finished looking for decompositions, didn't find any, and so now we have to convert all the inputs to meta and try to run the meta implementation of the function to try to figure out what the output looks like.

So, it sounds like the unfortunate series of events here, is that the meta implementation treats 1 element cpu tensors specially, but if we turned all the inputs into meta tensors, this logic no longer applies.

In other meta implementations, we have a concept of a `device_hint` which can be used to peek at the fake tensor device. Look in `torch/_meta_registrations.py` to see. This is probably the easy fix. Other ways to handle this are making sure you handle this logic in a decomposition, or do a direct implementation in fake tensor (since only direct implementations in fake tensor have access to cpu data.) To know which one to do requires knowing how we ended up here.

Reply:
Thank you! I'll use `device_hint` here